go/mysql/query.go

```diff
@@ -350,6 +350,10 @@ func (c *Conn) ReadQueryResult(maxrows int, wantfields bool) (result *sqltypes.R
 		}
 		switch data[0] {
 		case EOFPacket:
+			if len(data) != EOFPacketPayloadLen {
```

Comment:
IMO the style in the go driver seems cleaner than this switch, e.g. I think this is more straightforward:

```
if (data[0] == EOFPacket && len(data) == EOFPacketPayloadLen) {
    // This is what we expect.
    // Warnings and status flags are ignored.
    c.recycleReadPacket()
}
else if (data[0] == ErrPacket) {
    // Error packet.
    defer c.recycleReadPacket()
    return nil, ParseErrorPacket(data)
}
else {
    defer c.recycleReadPacket()
    return nil, fmt.Errorf("unexpected packet after fields: %v", data)
}
```

In fact, I think a utility function of `isEOFPacket(data)` that checks the first byte and the length makes this even more clear.

Reply:
Yep seems reasonable. There's still something strange going on based on my integration tests that I'll have to dig into.

Also there's this strange nugget of == 1: https://github.com/go-sql-driver/mysql/blob/master/packets.go#L649. No idea what's up there.